src/test/java/org/elasticsearch/discovery/zen/ZenDiscoveryTests.java

```diff
@@ -197,6 +197,7 @@ public void testNodeRejectsClusterStateWithWrongMasterNode() throws Exception {
         BytesStreamOutput bStream = new BytesStreamOutput();
         StreamOutput stream = CompressorFactory.defaultCompressor().streamOutput(bStream);
         stream.setVersion(node.version());
+        stream.writeBoolean(true); // To indicate that the full version is going to follow
```

Comment:
maybe make the PublishClusterStateAction#serializeFullClusterState  public ? we won't need to worry about this then.